src/sync/app_service_client.hpp

```diff
@@ -32,17 +32,29 @@ namespace app {
 class AppServiceClient {
 public:
     
-    /// Calls the MongoDB Stitch function with the provided name and arguments.
-    /// @param name The name of the Stitch function to be called.
+    virtual ~AppServiceClient() = default;
+
+    /// Calls the Realm Cloud function with the provided name and arguments.
+    /// @param name The name of the Realm Cloud function to be called.
     /// @param args The `BSONArray` of arguments to be provided to the function.
     /// @param service_name The name of the service, this is optional.
     /// @param completion_block Returns the result from the intended call, will return an Optional AppError is an error is thrown and a json string if successful
     virtual void call_function(const std::string& name,
                                const std::string& args_json,
                                const util::Optional<std::string>& service_name,
-                               std::function<void (util::Optional<AppError>, util::Optional<std::string>)> completion_block) = 0;
+                               std::function<void (util::Optional<AppError>, util::Optional<std::string>)> completion_block) const = 0;
     
-    virtual ~AppServiceClient() = default;
+    /// Calls the Realm Cloud function with the provided name and arguments.
+    /// @param name The name of the Realm Cloud function to be called.
+    /// @param args The `BSONArray` of arguments to be provided to the function.
+    /// @param completion_block Returns the result from the intended call, will return an Optional AppError is an error is thrown and a json string if successful
+    virtual void call_function(const std::string& name,
+                               const std::string& args_json,
+                               std::function<void (util::Optional<AppError>, util::Optional<std::string>)> completion_block) const = 0;
+    
+private:
+    /// The name of the Realm Cloud service
+    util::Optional<std::string> m_service_name;
```

Comment:
This member can't ever be used anywhere since this type doesn't have any non-pure-virtual functions.

Reply:
AppServiceClient is no longer abstract